### ISSUE TITLE
Compilation failing at first error

### ISSUE THREAD

**Issue by sergiofigueras** [google/error-prone#436]
Hello guys,

I've developed a BugChecker here but when I compile it fails at the first error, but there are other instances of that error that I would like to show to the user.

I mean:

```
/Users/sergio/Documents/projects/utqg-errorprone/utqg-errorprone-sampleusage/src/test/java/com/aurea/utqg/errorprone/sample/HelloNotPassing1Test.java:10: error: [UTQG:HelperMethodCannotBePublic] Helper Methods of test classes cannot be public.
  public void prepareMockedData() {
              ^
    (see https://www.aurea.com/utqg/HelperMethodCannotBePublic)
1 error
```

But there are a lot of UTQG:HelperMethodCannotBePublic errors that I would like to show to the user. In the actual process, user must fix each of them separately because they are displayed separately. I also tried mvn -fae -fn options but without any success.

How can I handle it?

**Comment by cushon** [comment]
It should show multiple errors within the same file, but the default behaviour is to stop after the first file with errors. If you want to get the full list of diagnostics, `-Xep:HelperMethodCannotBePublic:WARN` would allow compilation to continue past the first error.

**Comment by sergiofigueras** [comment]
Hello @cushon :),

We are triggering our build using teamcity, and the developers should look at the results and see what's going on and how to fix it. But the fact is that:
- we should display all errors, so developer can fix all errors instead of triggering different builds to do this.
- the categorization of errors that errorprone provides (warning, error, etc) should be used in that case. So, if a BugChecker that's marked as error is described, it should fail the build.

How do you think that I could handle that? Should I specialize the compiler?

**Comment by regrog** [comment]
As cushon said, you should change errors gravity to WARN for each error you got.
I did the same in my build the first time I used error prone.
I changed all these defaults and checked all of them at the end of the build process.
You could have more errors, just add the error type each time the build fails.
When you have done fixing all of them, remove every gravity override.
```
-Xep:SelfAssignment:WARN
-Xep:ReturnValueIgnored:WARN
-Xep:SizeGreaterThanOrEqualsZero:WARN
-Xep:ArrayToString:WARN
-Xep:DepAnn:WARN
-Xep:ComparisonOutOfRange:WARN
-Xep:GetClassOnClass:WARN
-Xep:FormatString:WARN
-Xep:DeadException:WARN
```

**Comment by vanniktech** [comment]
#424 and the suggestion form @cushon would come in handy here

**Comment by PhilippWendler** [comment]
We would also like a way to achieve this without marking errors as warnings. In fact, we use `-Werror` in our build on purpose to fail the build if error-prone reports anything. Now it would be nice if the reports from our CI build would include all problems and not only those from one file. Marking errors as warnings would mean the CI build would not be marked as failure at all, so this is not a solution.

**Comment by cpovirk** [comment]
I think we think this should have been fixed for most tools [since Error Prone 2.3.2](https://github.com/google/error-prone/commit/91a57840f42c887a55ccc38cf1fb6aa4f5f6f111) (mentioned in https://github.com/google/error-prone/issues/1341#issuecomment-526546913) and that it will be fixed for Bazel [in 8.0.0](https://github.com/bazelbuild/bazel/commit/242adde185ba10799d03be228466943750e1d5e9).

**Comment by cpovirk** [comment]
Coincidentally, I saw some evidence today that `-Werror` does _not_ output errors from multiple files with at least _built-in_ javac warnings (as of JDK 22). It's possible that Error Prone warnings would suffer from the same problem in combination with `-Werror` nowadays, but I haven't checked. Actual Error Prone _errors_ should at least be fixed as outlined in my previous post, even under JDK 22, I hope. (It works for us with an internal Bazel and an even newer javac, at least in the tests I've done.)

**Comment by PavelTurk** [comment]
I have the same problem this is my config

```
<plugin>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>${mave-compiler-plugin.version}</version>
    <configuration>
        <release>${maven.compiler.release}</release>
        <parameters>true</parameters>
        <encoding>UTF-8</encoding>
        <fork>true</fork>
        <showWarnings>true</showWarnings>
        <compilerArgs>
            <arg>-Xlint:rawtypes</arg>
            <arg>-Xlint:unchecked</arg>
            <arg>-XDcompilePolicy=simple</arg>
            <arg>--should-stop=ifError=FLOW</arg>
            <arg>-Werror</arg>
            <arg>-Xplugin:ErrorProne -XepAllErrorsAsWarnings -Xep:MissingSummary:OFF</arg>

            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
            <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
        </compilerArgs>
        <annotationProcessorPaths>
            <path>
                <groupId>com.google.errorprone</groupId>
                <artifactId>error_prone_core</artifactId>
                <version>2.48.0</version>
            </path>
        </annotationProcessorPaths>
    </configuration>
</plugin>
```
I want Error Prone to show all errors in the module before stopping. Currently it only shows errors from one file and then stops. So I compile, fix the errors in that file, compile again, get errors from another file, and so on.
